11-functions.Rmd

```diff
@@ -5,6 +5,121 @@
 ottrpal::set_knitr_image_path()
 ```
 
-```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Major point!! example image"}
+```{r, fig.align='center', fig.alt = "Learning objectives are to be able to: Recognize when it is appropriate to make a custom function to aid in the reproducibility of an analysis. Create functions that are reproducible and useable by others. What aspects of a function make them useable by others. Well documented. Follow a style. Properly engineered", out.width="100%", echo = FALSE}
 ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g21a84b32106_0_53")
 ```
+
+### Recognize when to write a custom function
+
+There's a lot of reasons writing your own custom functions might be a thing you need to do. Here we will discuss two of the major reasons.
+
+#### Cutting down on repetitive code
+
+We've discussed how DRY code is easier to maintain and understand! It is easier to fix code in one place as opposed to three or more places. A general guideline is that if you need two use a same chunk of code more than twice you probably should invest the time to make a custom function for it.
```

Comment:
Might be good to put a dictionary box for people in case they forget what DRY code means... if we don't already have one from the previous chapters maybe we can put one in both places?

Reply:
I made one below... I can't recall If I added this to the style so I will check that

Reply:
How about linking to that section? I think I like that better?